Fixing compiler warnings

JurgenLB opened the pull request:
With the help off copilot I have solved some compiling Warnings;

```
main/unzip_iterator.h:53:50: warning: 'template<class _Category, class _Tp, class _Distance, class _Pointer, class _Reference> struct std::iterator' is deprecated [-Wdeprecated-declarations]
   53 |         class basic_unzip_iterator : public std::iterator<std::input_iterator_tag, basic_unzip_stream<CharT, Traits> > {
      |                                                  ^~~~~~~~
In file included from /usr/include/c++/12/string:45,
                 from /home/lauwers/domoticz/main/stdafx.h:18,
                 from /home/lauwers/domoticz/CMakeFiles/domoticz.dir/cmake_pch.hxx:5,
                 from <command-line>:
/usr/include/c++/12/bits/stl_iterator_base_types.h:127:34: note: declared here
  127 |     struct _GLIBCXX17_DEPRECATED iterator
```

```
warning: ‘%02d’ directive output may be truncated writing 2 bytes into a region of size between 0 and 16 [-Wformat-truncation=]
  249 |         snprintf(szStartTime, sizeof(szStartTime), "%04d-%02d-%02d %02d:%02d:%02d", last_hour.tm_year + 1900, last_hour.tm_mon + 1, last_hour.tm_mday, last_hour.tm_hour, 0, 0);
      |                                                                         ^~~~
In file included from /usr/include/stdio.h:894,
                 from /home/runner/work/domoticz/domoticz/main/stdafx.h:16,
                 from /home/runner/work/domoticz/domoticz/CMakeFiles/domoticz.dir/cmake_pch.hxx:5,
                 from <command-line>:
/usr/include/x86_64-linux-gnu/bits/stdio2.h:71:35: note: ‘__builtin___snprintf_chk’ output between 20 and 54 bytes into a destination of size 30
   71 |   return __builtin___snprintf_chk (__s, __n, __USE_FORTIFY_LEVEL - 1,
      |          ~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   72 |                                    __glibc_objsize (__s), __fmt,
      |                                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   73 |                                    __va_arg_pack ());
      |                                    ~~~~~~~~~~~~~~~~~
```

There is still a Warning from JSON-Reader

Kind Regards

JurgenLB committed:
Replace deprecated std::iterator inheritance

Updated the last modified date and improved the basic_unzip_iterator class by replacing deprecated std::iterator inheritance with explicit typedefs and adding a shared pointer for the stream object.

JurgenLB committed:
Refactor comments in unzip_iterator.h

Updated comments for clarity and formatting.

JurgenLB committed:
Fixing snprintf warning with strftime usage

Updated szStartTime buffer size and changed time formatting to use strftime for better readability and error handling.

gizmocuz commented:
I don't have this main/KWHStats.cpp warning. Also pretty sure it should ever fail, but increasing the buffer was probably enough?

Also no warnings for the JSON reader,  but I do notice some warnings in the JWT token helper functions

JurgenLB commented:
I will try with only changing the size to 32.

I see this in the Domoticz PR Check / build(pull_request)

```
Building CXX object extern/jsoncpp/src/lib_json/CMakeFiles/jsoncpp_static.dir/json_reader.cpp.o
/home/runner/work/domoticz/domoticz/extern/jsoncpp/src/lib_json/json_reader.cpp:756:34: warning: ‘Reader’ is deprecated: Use CharReader and CharReaderBuilder instead. [-Wdeprecated-declarations]
  756 | Reader::Char Reader::getNextChar() {
      |                                  ^
In file included from /home/runner/work/domoticz/domoticz/extern/jsoncpp/src/lib_json/json_reader.cpp:10:
/home/runner/work/domoticz/domoticz/extern/jsoncpp/include/json/reader.h:37:63: note: declared here
   37 |     "Use CharReader and CharReaderBuilder instead.") JSON_API Reader {
      |                                                               ^~~~~~
/home/runner/work/domoticz/domoticz/extern/jsoncpp/src/lib_json/json_reader.cpp:810:21: warning: ‘Reader’ is deprecated: Use CharReader and CharReaderBuilder instead. [-Wdeprecated-declarations]
  810 | std::vector<Reader::StructuredError> Reader::getStructuredErrors() const {
      |                     ^~~~~~~~~~~~~~~
/home/runner/work/domoticz/domoticz/extern/jsoncpp/include/json/reader.h:37:63: note: declared here
   37 |     "Use CharReader and CharReaderBuilder instead.") JSON_API Reader {
      |                                                               ^~~~~~
```

JurgenLB committed:
Simplify date formatting in KWHStats

gizmocuz commented:
Hmmm interesting.... We use the jsoncpp library and these files are hosted remotely on their repository.
Could it be that it's not correct there?

https://github.com/open-source-parsers/jsoncpp/blob/master/src/lib_json/json_reader.cpp

Or is your system using out of date libraries?

could you try this in your dev folder?

git submodule update --init --remote --recursive

(do note that jsoncpp latest release tag is 1.9.6 from Sep 10 2024)

gizmocuz commented:
Travis mentiones this about the time string

 output between 20 and 54 bytes into a destination of size 32

JurgenLB commented:
For the jsoncpp library this was the output on Github and on a system with a clean install from GitHub .
But I will try the submodule update

Only changing the size, dit not resolve this.

```
/home/runner/work/domoticz/domoticz/main/KWHStats.cpp: In static member function ‘static void CKWHStats::HandleKWHStatsHour()’:
/home/runner/work/domoticz/domoticz/main/KWHStats.cpp:249:77: warning: ‘:’ directive output may be truncated writing 1 byte into a region of size between 0 and 16 [-Wformat-truncation=]
  249 |         snprintf(szStartTime, sizeof(szStartTime), "%04d-%02d-%02d %02d:%02d:%02d", last_hour.tm_year + 1900, last_hour.tm_mon + 1, last_hour.tm_mday, last_hour.tm_hour, 0, 0);
      |                                                                             ^
In file included from /usr/include/stdio.h:980,
                 from /home/runner/work/domoticz/domoticz/main/stdafx.h:16,
                 from /home/runner/work/domoticz/domoticz/CMakeFiles/domoticz.dir/cmake_pch.hxx:5,
                 from <command-line>:
In function ‘int snprintf(char*, size_t, const char*, ...)’,
    inlined from ‘static void CKWHStats::HandleKWHStatsHour()’ at /home/runner/work/domoticz/domoticz/main/KWHStats.cpp:249:10:
/usr/include/x86_64-linux-gnu/bits/stdio2.h:54:35: note: ‘__builtin___snprintf_chk’ output between 20 and 54 bytes into a destination of size 32
   54 |   return __builtin___snprintf_chk (__s, __n, __USE_FORTIFY_LEVEL - 1,
      |          ~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   55 |                                    __glibc_objsize (__s), __fmt,
      |                                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   56 |                                    __va_arg_pack ());
      |                                    ~~~~~~~~~~~~~~~~~
```

JurgenLB committed:
Refactor time formatting in KWHStats.cpp

Updated time formatting to use strftime for better readability and added a fallback for error handling.

JurgenLB commented:
I did ;
`git pull`
`rm -r extern/jsoncpp/`
`cmake -DCMAKE_BUILD_TYPE=Release CMakeLists.txt`
-- Submodule update
Submodule path 'extern/jsoncpp': checked out '2a6e163b02ac5044d74a44ed15f8aa7af687aea5'
`git submodule update --init --remote --recursive`
`make`
Warning;
```
Building CXX object extern/jsoncpp/src/lib_json/CMakeFiles/jsoncpp_static.dir/json_reader.cpp.o
/home/lauwers/domoticz/extern/jsoncpp/src/lib_json/json_reader.cpp:756:34: warning: â€Readerâ€ is deprecated: Use CharReader and CharReaderBuilder instead. [-Wdeprecated-declarations]
  756 | Reader::Char Reader::getNextChar() {
      |                                  ^
In file included from /home/lauwers/domoticz/extern/jsoncpp/src/lib_json/json_reader.cpp:10:
/home/lauwers/domoticz/extern/jsoncpp/include/json/reader.h:37:63: note: declared here
   37 |     "Use CharReader and CharReaderBuilder instead.") JSON_API Reader {
      |                                                               ^~~~~~
/home/lauwers/domoticz/extern/jsoncpp/src/lib_json/json_reader.cpp:810:21: warning: â€Readerâ€ is deprecated: Use CharReader and CharReaderBuilder instead. [-Wdeprecated-declarations]
  810 | std::vector<Reader::StructuredError> Reader::getStructuredErrors() const {
      |                     ^~~~~~~~~~~~~~~
/home/lauwers/domoticz/extern/jsoncpp/include/json/reader.h:37:63: note: declared here
   37 |     "Use CharReader and CharReaderBuilder instead.") JSON_API Reader {
      |                                                               ^~~~~~
```

When I do a new "make" command the warning from the submodule is not visible
But this is logical because there is no change from the submodule I believe 

Kind Regards

JurgenLB commented:
the JSON warning can also be silenced;
```
# It silences deprecated-declaration warnings for non-MSVC compilers
# and the equivalent warning for MSVC.
if (TARGET jsoncpp_static)
  if (MSVC)
    target_compile_options(jsoncpp_static PRIVATE /wd4996)
  else()
    target_compile_options(jsoncpp_static PRIVATE -Wno-deprecated-declarations)
  endif()
endif()
```
But also the FindBoost function give a warning;
```
CMake Warning (dev) at CMakeLists.txt:688 (find_package):
  Policy CMP0167 is not set: The FindBoost module is removed.  Run "cmake
  --help-policy CMP0167" for policy details.  Use the cmake_policy command to
  set the policy and suppress this warning.

This warning is for project developers.  Use -Wno-dev to suppress it.
```

gizmocuz commented:
Thank you for this PR!

JurgenLB commented:
the Deprecation warnings are from;

```
// ‘Reader’ is deprecated: Use CharReader and CharReaderBuilder instead.
Reader::Char Reader::getNextChar() {
  if (current_ == end_)
    return 0;
  return *current_++;
}
```
but this is used multiple times.

and

```
// ‘Reader’ is deprecated: Use CharReader and CharReaderBuilder instead.
//std::vector<Reader::StructuredError> Reader::getStructuredErrors() const {
//  std::vector<Reader::StructuredError> allErrors;
//  for (const auto& error : errors_) {
//    Reader::StructuredError structured;
//    structured.offset_start = error.token_.start_ - begin_;
//    structured.offset_limit = error.token_.end_ - begin_;
//    structured.message = error.message_;
//    allErrors.push_back(structured);
//  }
//  return allErrors;
//}
```

this last I can comment out and domoticz still build.

gizmocuz commented:
From which file are these warnings?

JurgenLB commented:
it is from `extern/jsoncpp/src/lib_json/json_reader.cpp`
line 756
and 
line 810

gizmocuz commented:
Yes, as mentioned, this is in the original library. Maybe you could make a git issue on their project page for this...
Seeing this issue for months and months now myself

JurgenLB commented:
I can confirm that with the latest jsoncpp version 1.9.7 (b511d9e) the warnings from json_reader are gone.

But I get a lot of errors on `undefined reference to `Json::Value::`
even with copilot I found no solution for the moment.

JurgenLB commented:
it seems that Jsoncpp sets CMAKE_CXX_STANDARD to 11, which doesn't have std::string_view. This is the root cause of the linking errors.
Domoticz is set to Cmake 17

JurgenLB commented:
I now have a clean build.
But I think you don't want all the changes in a PR.

I can make a PR if you wan't, let me know your preferences.

[https://github.com/JurgenLB/domoticz/pull/14](url)

```
Run cd $GITHUB_WORKSPACE
-- The C compiler identification is GNU 13.3.0
-- The CXX compiler identification is GNU 13.3.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Submodule update
Submodule 'extern/fmtlib' (https://github.com/fmtlib/fmt.git) registered for path 'extern/fmtlib'
Submodule 'extern/jsoncpp' (https://github.com/open-source-parsers/jsoncpp.git) registered for path 'extern/jsoncpp'
Submodule 'extern/jwtcpp' (https://github.com/Thalhammer/jwt-cpp) registered for path 'extern/jwtcpp'
Submodule 'minizip' (https://github.com/domoticz/minizip.git) registered for path 'extern/minizip'
Submodule 'extern/mosquitto' (https://github.com/eclipse/mosquitto.git) registered for path 'extern/mosquitto'
Submodule 'extern/sqlite-amalgamation' (https://github.com/azadkuh/sqlite-amalgamation.git) registered for path 'extern/sqlite-amalgamation'
Cloning into '/home/runner/work/domoticz/domoticz/extern/fmtlib'...
Cloning into '/home/runner/work/domoticz/domoticz/extern/jsoncpp'...
Cloning into '/home/runner/work/domoticz/domoticz/extern/jwtcpp'...
Cloning into '/home/runner/work/domoticz/domoticz/extern/minizip'...
Cloning into '/home/runner/work/domoticz/domoticz/extern/mosquitto'...
Cloning into '/home/runner/work/domoticz/domoticz/extern/sqlite-amalgamation'...
Submodule path 'extern/fmtlib': checked out '91d1aced0a846161135310249664506b4a56e196'
Submodule path 'extern/jsoncpp': checked out 'b511d9e64956db998b74909df112ac8c8f41d6ff'
Submodule path 'extern/jwtcpp': checked out '7fc762d7d6b6e5df875206497966f23eda5da229'
Submodule path 'extern/minizip': checked out 'aee7fbddf118d9363575af96310c29fa747d70c0'
Submodule path 'extern/mosquitto': checked out '28f914788f6a23781d724d31a4d44e2616815964'
Submodule path 'extern/sqlite-amalgamation': checked out '15d0ff10ebc7e7225eced1de84bb52137000899b'
-- Applying jsoncpp patch for domoticz
-- LUA library found at: /usr/lib/x86_64-linux-gnu/liblua5.3.a
-- Found Python3: /usr/include/python3.12 (found suitable version "3.12.3", minimum required is "3.4") found components: Development Development.Module Development.Embed
-- Python3 version 3.12.3 includes found at: /usr/include/python3.12
-- Looking for execinfo.h
-- Looking for execinfo.h - found
-- Looking for 3 include files sys/types.h, ..., linux/i2c.h
-- Looking for 3 include files sys/types.h, ..., linux/i2c.h - found
-- Building with I2C support
-- Looking for include files sys/types.h, linux/spi/spidev.h
-- Looking for include files sys/types.h, linux/spi/spidev.h - found
-- Building with SPI support
-- ###########################
-- Compiling Revision #17017
-- ###########################
-- Using builtin jsoncpp library
-- JsonCpp Version: 1.9.7
-- Looking for memset_s
-- Looking for memset_s - not found
-- Looking for C++ include clocale
-- Looking for C++ include clocale - found
-- Looking for localeconv
-- Looking for localeconv - found
-- Looking for C++ include sys/types.h
-- Looking for C++ include sys/types.h - found
-- Looking for C++ include stdint.h
-- Looking for C++ include stdint.h - found
-- Looking for C++ include stddef.h
-- Looking for C++ include stddef.h - found
-- Check size of lconv
-- Check size of lconv - done
-- Performing Test HAVE_DECIMAL_POINT
-- Performing Test HAVE_DECIMAL_POINT - Success
-- Using builtin Mosquitto library
-- Found OpenSSL: /usr/lib/x86_64-linux-gnu/libcrypto.so (found version "3.0.13")
-- WITH_DLT = OFF
-- Could NOT find cJSON (missing: CJSON_INCLUDE_DIR CJSON_LIBRARY) 
-- Optional dependency cJSON not found. Some features will be disabled.
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE
-- Looking for getaddrinfo_a in anl
-- Looking for getaddrinfo_a in anl - found
-- Found SQLite3: /usr/include (found version "3.45.1")
-- SQLite library found at: /usr/lib/x86_64-linux-gnu/libsqlite3.so
-- SQLite includes found at: /usr/include
-- Found ZLIB: /usr/lib/x86_64-linux-gnu/libz.so (found version "1.3")
-- Using builtin jwt-cpp library
-- OPENSSL library found at: /usr/lib/x86_64-linux-gnu/libssl.so;/usr/lib/x86_64-linux-gnu/libcrypto.so
-- Linking against boost static libraries
-- BOOST 1.86.0 includes found at: 
-- ZLIB library found at: /usr/lib/x86_64-linux-gnu/libz.so
-- ZLIB includes found at: /usr/include
-- Found CURL: /usr/lib/x86_64-linux-gnu/libcurl.so (found version "8.5.0")
-- Curl library found at: /usr/lib/x86_64-linux-gnu/libcurl.so
-- Curl includes found at: /usr/include/x86_64-linux-gnu
-- Found LIBUSB: /usr/lib/x86_64-linux-gnu/libusb.so
-- LIBUSB library found at: /usr/lib/x86_64-linux-gnu/libusb.so
-- OpenZWave library found at: /home/runner/work/domoticz/open-zwave-read-only/libopenzwave.a
-- OpenZWave includes found at: /home/runner/work/domoticz/open-zwave-read-only/cpp/src
-- GPIO is available
-- Looking for res_init
-- Looking for res_init - not found
-- Using precompiled headers
-- Using static libgcc/libstdc++
-- Configuring done (9.3s)
-- Generating done (0.1s)
-- Build files have been written to: /home/runner/work/domoticz/domoticz
[  0%] Building C object extern/minizip/CMakeFiles/minizip.dir/ioapi.c.o
[  0%] Building C object extern/minizip/CMakeFiles/minizip.dir/unzip.c.o
[  1%] Building C object extern/minizip/CMakeFiles/minizip.dir/zip.c.o
[  1%] Linking C static library libminizip.a
[  1%] Built target minizip
-- Found Git: /usr/bin/git (found version "2.52.0")
[  1%] Built target revisiontag
[  1%] Building CXX object extern/jsoncpp/src/lib_json/CMakeFiles/jsoncpp_static.dir/json_reader.cpp.o
[  1%] Building CXX object extern/jsoncpp/src/lib_json/CMakeFiles/jsoncpp_static.dir/json_value.cpp.o
[  2%] Building CXX object extern/jsoncpp/src/lib_json/CMakeFiles/jsoncpp_static.dir/json_writer.cpp.o
[  2%] Linking CXX static library libjsoncpp.a
[  2%] Built target jsoncpp_static
[  2%] Building C object extern/mosquitto/lib/CMakeFiles/libmosquitto_static.dir/actions.c.o
[  2%] Building C object extern/mosquitto/lib/CMakeFiles/libmosquitto_static.dir/callbacks.c.o
[  2%] Building C object extern/mosquitto/lib/CMakeFiles/libmosquitto_static.dir/connect.c.o
[  4%] Building C object extern/mosquitto/lib/CMakeFiles/libmosquitto_static.dir/handle_auth.c.o
[  4%] Building C object extern/mosquitto/lib/CMakeFiles/libmosquitto_static.dir/handle_connack.c.o
[  4%] Building C object extern/mosquitto/lib/CMakeFiles/libmosquitto_static.dir/handle_disconnect.c.o
[  4%] Building C object extern/mosquitto/lib/CMakeFiles/libmosquitto_static.dir/handle_ping.c.o
[  5%] Building C object extern/mosquitto/lib/CMakeFiles/libmosquitto_static.dir/handle_pubackcomp.c.o
[  5%] Building C object extern/mosquitto/lib/CMakeFiles/libmosquitto_static.dir/handle_publish.c.o
[  5%] Building C object extern/mosquitto/lib/CMakeFiles/libmosquitto_static.dir/handle_pubrec.c.o
[  5%] Building C object extern/mosquitto/lib/CMakeFiles/libmosquitto_static.dir/handle_pubrel.c.o
[  5%] Building C object extern/mosquitto/lib/CMakeFiles/libmosquitto_static.dir/handle_suback.c.o
[  7%] Building C object extern/mosquitto/lib/CMakeFiles/libmosquitto_static.dir/handle_unsuback.c.o
[  7%] Building C object extern/mosquitto/lib/CMakeFiles/libmosquitto_static.dir/helpers.c.o
[  7%] Building C object extern/mosquitto/lib/CMakeFiles/libmosquitto_static.dir/logging_mosq.c.o
[  7%] Building C object extern/mosquitto/lib/CMakeFiles/libmosquitto_static.dir/loop.c.o
[  7%] Building C object extern/mosquitto/lib/CMakeFiles/libmosquitto_static.dir/memory_mosq.c.o
[  8%] Building C object extern/mosquitto/lib/CMakeFiles/libmosquitto_static.dir/messages_mosq.c.o
[  8%] Building C object extern/mosquitto/lib/CMakeFiles/libmosquitto_static.dir/misc_mosq.c.o
[  8%] Building C object extern/mosquitto/lib/CMakeFiles/libmosquitto_static.dir/mosquitto.c.o
[  8%] Building C object extern/mosquitto/lib/CMakeFiles/libmosquitto_static.dir/net_mosq_ocsp.c.o
[  8%] Building C object extern/mosquitto/lib/CMakeFiles/libmosquitto_static.dir/net_mosq.c.o
[ 10%] Building C object extern/mosquitto/lib/CMakeFiles/libmosquitto_static.dir/options.c.o
[ 10%] Building C object extern/mosquitto/lib/CMakeFiles/libmosquitto_static.dir/packet_datatypes.c.o
[ 10%] Building C object extern/mosquitto/lib/CMakeFiles/libmosquitto_static.dir/packet_mosq.c.o
[ 10%] Building C object extern/mosquitto/lib/CMakeFiles/libmosquitto_static.dir/property_mosq.c.o
[ 10%] Building C object extern/mosquitto/lib/CMakeFiles/libmosquitto_static.dir/read_handle.c.o
[ 11%] Building C object extern/mosquitto/lib/CMakeFiles/libmosquitto_static.dir/send_connect.c.o
[ 11%] Building C object extern/mosquitto/lib/CMakeFiles/libmosquitto_static.dir/send_disconnect.c.o
[ 11%] Building C object extern/mosquitto/lib/CMakeFiles/libmosquitto_static.dir/send_mosq.c.o
[ 11%] Building C object extern/mosquitto/lib/CMakeFiles/libmosquitto_static.dir/send_publish.c.o
[ 11%] Building C object extern/mosquitto/lib/CMakeFiles/libmosquitto_static.dir/send_subscribe.c.o
[ 13%] Building C object extern/mosquitto/lib/CMakeFiles/libmosquitto_static.dir/send_unsubscribe.c.o
[ 13%] Building C object extern/mosquitto/lib/CMakeFiles/libmosquitto_static.dir/socks_mosq.c.o
[ 13%] Building C object extern/mosquitto/lib/CMakeFiles/libmosquitto_static.dir/srv_mosq.c.o
[ 13%] Building C object extern/mosquitto/lib/CMakeFiles/libmosquitto_static.dir/strings_mosq.c.o
[ 13%] Building C object extern/mosquitto/lib/CMakeFiles/libmosquitto_static.dir/thread_mosq.c.o
[ 14%] Building C object extern/mosquitto/lib/CMakeFiles/libmosquitto_static.dir/time_mosq.c.o
[ 14%] Building C object extern/mosquitto/lib/CMakeFiles/libmosquitto_static.dir/tls_mosq.c.o
[ 14%] Building C object extern/mosquitto/lib/CMakeFiles/libmosquitto_static.dir/utf8_mosq.c.o
[ 14%] Building C object extern/mosquitto/lib/CMakeFiles/libmosquitto_static.dir/util_mosq.c.o
[ 16%] Building C object extern/mosquitto/lib/CMakeFiles/libmosquitto_static.dir/util_topic.c.o
[ 16%] Building C object extern/mosquitto/lib/CMakeFiles/libmosquitto_static.dir/will_mosq.c.o
[ 16%] Linking C static library libmosquitto_static.a
[ 16%] Built target libmosquitto_static
[ 16%] Building CXX object CMakeFiles/domoticz.dir/cmake_pch.hxx.gch
[ 16%] Building CXX object CMakeFiles/domoticz.dir/main/stdafx.cpp.o
[ 16%] Building CXX object CMakeFiles/domoticz.dir/main/Alexa.cpp.o
[ 16%] Building CXX object CMakeFiles/domoticz.dir/main/BaroForecastCalculator.cpp.o
[ 17%] Building CXX object CMakeFiles/domoticz.dir/main/CmdLine.cpp.o
[ 17%] Building CXX object CMakeFiles/domoticz.dir/main/Camera.cpp.o
[ 17%] Building CXX object CMakeFiles/domoticz.dir/main/domoticz.cpp.o
[ 17%] Building CXX object CMakeFiles/domoticz.dir/main/dzVents.cpp.o
[ 17%] Building CXX object CMakeFiles/domoticz.dir/main/EventSystem.cpp.o
[ 19%] Building CXX object CMakeFiles/domoticz.dir/main/EventsPythonModule.cpp.o
[ 19%] Building CXX object CMakeFiles/domoticz.dir/main/EventsPythonDevice.cpp.o
[ 19%] Building CXX object CMakeFiles/domoticz.dir/main/Helper.cpp.o
[ 19%] Building CXX object CMakeFiles/domoticz.dir/main/HTMLSanitizer.cpp.o
[ 19%] Building CXX object CMakeFiles/domoticz.dir/main/IFTTT.cpp.o
[ 20%] Building CXX object CMakeFiles/domoticz.dir/main/json_helper.cpp.o
[ 20%] Building CXX object CMakeFiles/domoticz.dir/main/KWHStats.cpp.o
[ 20%] Building CXX object CMakeFiles/domoticz.dir/main/localtime_r.cpp.o
[ 20%] Building CXX object CMakeFiles/domoticz.dir/main/lsignal.cpp.o
[ 20%] Building CXX object CMakeFiles/domoticz.dir/main/Logger.cpp.o
[ 22%] Building CXX object CMakeFiles/domoticz.dir/main/LuaCommon.cpp.o
[ 22%] Building CXX object CMakeFiles/domoticz.dir/main/LuaHandler.cpp.o
[ 22%] Building CXX object CMakeFiles/domoticz.dir/main/LuaTable.cpp.o
[ 22%] Building CXX object CMakeFiles/domoticz.dir/main/mainworker.cpp.o
[ 22%] Building CXX object CMakeFiles/domoticz.dir/main/mosquitto_helper.cpp.o
[ 23%] Building CXX object CMakeFiles/domoticz.dir/main/NotificationObserver.cpp.o
[ 23%] Building CXX object CMakeFiles/domoticz.dir/main/NotificationSystem.cpp.o
[ 23%] Building CXX object CMakeFiles/domoticz.dir/main/RFXNames.cpp.o
[ 23%] Building CXX object CMakeFiles/domoticz.dir/main/Scheduler.cpp.o
[ 23%] Building CXX object CMakeFiles/domoticz.dir/main/SignalHandler.cpp.o
[ 25%] Building CXX object CMakeFiles/domoticz.dir/main/SQLHelper.cpp.o
[ 25%] Building CXX object CMakeFiles/domoticz.dir/main/StoppableTask.cpp.o
[ 25%] Building CXX object CMakeFiles/domoticz.dir/main/SunRiseSet.cpp.o
[ 25%] Building CXX object CMakeFiles/domoticz.dir/main/TrendCalculator.cpp.o
[ 25%] Building CXX object CMakeFiles/domoticz.dir/main/WebServer.cpp.o
[ 26%] Building CXX object CMakeFiles/domoticz.dir/main/WebServerCmds.cpp.o
[ 26%] Building CXX object CMakeFiles/domoticz.dir/main/WebServerCommands.cpp.o
[ 26%] Building CXX object CMakeFiles/domoticz.dir/main/WebServerHandleGraph.cpp.o
[ 26%] Building CXX object CMakeFiles/domoticz.dir/main/WebServerHelper.cpp.o
[ 27%] Building CXX object CMakeFiles/domoticz.dir/main/WindCalculation.cpp.o
[ 27%] Building CXX object CMakeFiles/domoticz.dir/mdns/mdns.cpp.o
[ 27%] Building CXX object CMakeFiles/domoticz.dir/iamserver/IamService.cpp.o
[ 27%] Building CXX object CMakeFiles/domoticz.dir/mcpserver/McpService.cpp.o
[ 27%] Building CXX object CMakeFiles/domoticz.dir/push/BasePush.cpp.o
[ 29%] Building CXX object CMakeFiles/domoticz.dir/push/FibaroPush.cpp.o
[ 29%] Building CXX object CMakeFiles/domoticz.dir/push/GooglePubSubPush.cpp.o
[ 29%] Building CXX object CMakeFiles/domoticz.dir/push/HttpPush.cpp.o
[ 29%] Building CXX object CMakeFiles/domoticz.dir/push/InfluxPush.cpp.o
[ 29%] Building CXX object CMakeFiles/domoticz.dir/push/MQTTPush.cpp.o
[ 30%] Building CXX object CMakeFiles/domoticz.dir/push/WebsocketPush.cpp.o
[ 30%] Building CXX object CMakeFiles/domoticz.dir/httpclient/HTTPClient.cpp.o
[ 30%] Building CXX object CMakeFiles/domoticz.dir/httpclient/UrlEncode.cpp.o
[ 30%] Building CXX object CMakeFiles/domoticz.dir/hardware/1Wire.cpp.o
[ 30%] Building CXX object CMakeFiles/domoticz.dir/hardware/1Wire/1WireByOWFS.cpp.o
[ 32%] Building CXX object CMakeFiles/domoticz.dir/hardware/1Wire/1WireByKernel.cpp.o
[ 32%] Building CXX object CMakeFiles/domoticz.dir/hardware/1Wire/1WireCommon.cpp.o
[ 32%] Building CXX object CMakeFiles/domoticz.dir/hardware/1Wire/1WireForWindows.cpp.o
[ 32%] Building CXX object CMakeFiles/domoticz.dir/hardware/AccuWeather.cpp.o
[ 32%] Building CXX object CMakeFiles/domoticz.dir/hardware/AirconWithMe.cpp.o
[ 33%] Building CXX object CMakeFiles/domoticz.dir/hardware/AlfenEve.cpp.o
[ 33%] Building CXX object CMakeFiles/domoticz.dir/hardware/AnnaThermostat.cpp.o
[ 33%] Building CXX object CMakeFiles/domoticz.dir/hardware/Arilux.cpp.o
[ 33%] Building CXX object CMakeFiles/domoticz.dir/hardware/ASyncSerial.cpp.o
[ 33%] Building CXX object CMakeFiles/domoticz.dir/hardware/ASyncTCP.cpp.o
[ 35%] Building CXX object CMakeFiles/domoticz.dir/hardware/AtagOne.cpp.o
[ 35%] Building CXX object CMakeFiles/domoticz.dir/hardware/BleBox.cpp.o
[ 35%] Building CXX object CMakeFiles/domoticz.dir/hardware/Buienradar.cpp.o
[ 35%] Building CXX object CMakeFiles/domoticz.dir/hardware/cayenne_lpp/CayenneLPP_Dec.cpp.o
[ 35%] Building CXX object CMakeFiles/domoticz.dir/hardware/ColorSwitch.cpp.o
[ 36%] Building CXX object CMakeFiles/domoticz.dir/hardware/Comm5Serial.cpp.o
[ 36%] Building CXX object CMakeFiles/domoticz.dir/hardware/Comm5SMTCP.cpp.o
[ 36%] Building CXX object CMakeFiles/domoticz.dir/hardware/Comm5TCP.cpp.o
[ 36%] Building CXX object CMakeFiles/domoticz.dir/hardware/CounterHelper.cpp.o
[ 36%] Building CXX object CMakeFiles/domoticz.dir/hardware/csocket.cpp.o
[ 38%] Building CXX object CMakeFiles/domoticz.dir/hardware/CurrentCostMeterBase.cpp.o
[ 38%] Building CXX object CMakeFiles/domoticz.dir/hardware/CurrentCostMeterSerial.cpp.o
[ 38%] Building CXX object CMakeFiles/domoticz.dir/hardware/CurrentCostMeterTCP.cpp.o
[ 38%] Building CXX object CMakeFiles/domoticz.dir/hardware/Daikin.cpp.o
[ 39%] Building CXX object CMakeFiles/domoticz.dir/hardware/DarkSky.cpp.o
[ 39%] Building CXX object CMakeFiles/domoticz.dir/hardware/DavisLoggerSerial.cpp.o
[ 39%] Building CXX object CMakeFiles/domoticz.dir/hardware/DenkoviDevices.cpp.o
[ 39%] Building CXX object CMakeFiles/domoticz.dir/hardware/DenkoviUSBDevices.cpp.o
[ 39%] Building CXX object CMakeFiles/domoticz.dir/hardware/DenkoviTCPDevices.cpp.o
[ 41%] Building CXX object CMakeFiles/domoticz.dir/hardware/DomoticzHardware.cpp.o
[ 41%] Building CXX object CMakeFiles/domoticz.dir/hardware/DomoticzInternal.cpp.o
[ 41%] Building CXX object CMakeFiles/domoticz.dir/hardware/DomoticzTCP.cpp.o
[ 41%] Building CXX object CMakeFiles/domoticz.dir/hardware/Dummy.cpp.o
[ 41%] Building CXX object CMakeFiles/domoticz.dir/hardware/EcoCompteur.cpp.o
[ 42%] Building CXX object CMakeFiles/domoticz.dir/hardware/EcoDevices.cpp.o
[ 42%] Building CXX object CMakeFiles/domoticz.dir/hardware/eHouseTCP.cpp.o
[ 42%] Building CXX object CMakeFiles/domoticz.dir/hardware/eHouse/EhouseEvents.cpp.o
[ 42%] Building CXX object CMakeFiles/domoticz.dir/hardware/eHouse/EhouseTcpClient.cpp.o
[ 42%] Building CXX object CMakeFiles/domoticz.dir/hardware/eHouse/EhouseUdpListener.cpp.o
[ 44%] Building CXX object CMakeFiles/domoticz.dir/hardware/Enever.cpp.o
[ 44%] Building CXX object CMakeFiles/domoticz.dir/hardware/EnOceanEEP.cpp.o
[ 44%] Building CXX object CMakeFiles/domoticz.dir/hardware/EnOceanESP2.cpp.o
[ 44%] Building CXX object CMakeFiles/domoticz.dir/hardware/EnOceanESP3.cpp.o
[ 44%] Building CXX object CMakeFiles/domoticz.dir/hardware/EnOceanRawValue.cpp.o
[ 45%] Building CXX object CMakeFiles/domoticz.dir/hardware/EnphaseAPI.cpp.o
[ 45%] Building CXX object CMakeFiles/domoticz.dir/hardware/Ec3kMeterTCP.cpp.o
[ 45%] Building CXX object CMakeFiles/domoticz.dir/hardware/eVehicles/TeslaApi.cpp.o
[ 45%] Building CXX object CMakeFiles/domoticz.dir/hardware/eVehicles/MercApi.cpp.o
[ 45%] Building CXX object CMakeFiles/domoticz.dir/hardware/eVehicles/TestcarApi.cpp.o
[ 47%] Building CXX object CMakeFiles/domoticz.dir/hardware/eVehicles/eVehicle.cpp.o
[ 47%] Building CXX object CMakeFiles/domoticz.dir/hardware/EvohomeBase.cpp.o
[ 47%] Building CXX object CMakeFiles/domoticz.dir/hardware/EvohomeRadio.cpp.o
[ 47%] Building CXX object CMakeFiles/domoticz.dir/hardware/EvohomeScript.cpp.o
[ 47%] Building CXX object CMakeFiles/domoticz.dir/hardware/EvohomeSerial.cpp.o
[ 48%] Building CXX object CMakeFiles/domoticz.dir/hardware/EvohomeTCP.cpp.o
[ 48%] Building CXX object CMakeFiles/domoticz.dir/hardware/EvohomeWeb.cpp.o
[ 48%] Building CXX object CMakeFiles/domoticz.dir/hardware/ETH8020.cpp.o
[ 48%] Building CXX object CMakeFiles/domoticz.dir/hardware/FritzboxTCP.cpp.o
[ 48%] Building CXX object CMakeFiles/domoticz.dir/hardware/Gpio.cpp.o
[ 50%] Building CXX object CMakeFiles/domoticz.dir/hardware/GpioPin.cpp.o
[ 50%] Building CXX object CMakeFiles/domoticz.dir/hardware/HardwareMonitor.cpp.o
[ 50%] Building CXX object CMakeFiles/domoticz.dir/hardware/HarmonyHub.cpp.o
[ 50%] Building CXX object CMakeFiles/domoticz.dir/hardware/Honeywell.cpp.o
[ 51%] Building CXX object CMakeFiles/domoticz.dir/hardware/HEOS.cpp.o
[ 51%] Building CXX object CMakeFiles/domoticz.dir/hardware/I2C.cpp.o
[ 51%] Building CXX object CMakeFiles/domoticz.dir/hardware/ICYThermostat.cpp.o
[ 51%] Building CXX object CMakeFiles/domoticz.dir/hardware/InComfort.cpp.o
[ 51%] Building CXX object CMakeFiles/domoticz.dir/hardware/KMTronicBase.cpp.o
[ 52%] Building CXX object CMakeFiles/domoticz.dir/hardware/KMTronic433.cpp.o
[ 52%] Building CXX object CMakeFiles/domoticz.dir/hardware/KMTronicSerial.cpp.o
[ 52%] Building CXX object CMakeFiles/domoticz.dir/hardware/KMTronicTCP.cpp.o
[ 52%] Building CXX object CMakeFiles/domoticz.dir/hardware/KMTronicUDP.cpp.o
[ 52%] Building CXX object CMakeFiles/domoticz.dir/hardware/Kodi.cpp.o
[ 54%] Building CXX object CMakeFiles/domoticz.dir/hardware/Limitless.cpp.o
[ 54%] Building CXX object CMakeFiles/domoticz.dir/hardware/LogitechMediaServer.cpp.o
[ 54%] Building CXX object CMakeFiles/domoticz.dir/hardware/Meteorologisk.cpp.o
[ 54%] Building CXX object CMakeFiles/domoticz.dir/hardware/Meteostick.cpp.o
[ 54%] Building CXX object CMakeFiles/domoticz.dir/hardware/MitsubishiWF.cpp.o
[ 55%] Building CXX object CMakeFiles/domoticz.dir/hardware/MochadTCP.cpp.o
[ 55%] Building CXX object CMakeFiles/domoticz.dir/hardware/MQTT.cpp.o
[ 55%] Building CXX object CMakeFiles/domoticz.dir/hardware/MQTTAutoDiscover.cpp.o
[ 55%] Building CXX object CMakeFiles/domoticz.dir/hardware/MultiFun.cpp.o
[ 55%] Building CXX object CMakeFiles/domoticz.dir/hardware/MySensorsBase.cpp.o
[ 57%] Building CXX object CMakeFiles/domoticz.dir/hardware/MySensorsSerial.cpp.o
[ 57%] Building CXX object CMakeFiles/domoticz.dir/hardware/MySensorsTCP.cpp.o
[ 57%] Building CXX object CMakeFiles/domoticz.dir/hardware/MySensorsMQTT.cpp.o
[ 57%] Building CXX object CMakeFiles/domoticz.dir/hardware/NefitEasy.cpp.o
[ 57%] Building CXX object CMakeFiles/domoticz.dir/hardware/Nest.cpp.o
[ 58%] Building CXX object CMakeFiles/domoticz.dir/hardware/NestOAuthAPI.cpp.o
[ 58%] Building CXX object CMakeFiles/domoticz.dir/hardware/Netatmo.cpp.o
[ 58%] Building CXX object CMakeFiles/domoticz.dir/hardware/HttpPoller.cpp.o
[ 58%] Building CXX object CMakeFiles/domoticz.dir/hardware/OctoPrintMQTT.cpp.o
[ 58%] Building CXX object CMakeFiles/domoticz.dir/hardware/OnkyoAVTCP.cpp.o
[ 60%] Building CXX object CMakeFiles/domoticz.dir/hardware/OpenWeatherMap.cpp.o
[ 60%] Building CXX object CMakeFiles/domoticz.dir/hardware/OpenWebNetTCP.cpp.o
[ 60%] Building CXX object CMakeFiles/domoticz.dir/hardware/OpenWebNetUSB.cpp.o
[ 60%] Building CXX object CMakeFiles/domoticz.dir/hardware/openwebnet/bt_openwebnet.cpp.o
[ 61%] Building CXX object CMakeFiles/domoticz.dir/hardware/OpenZWave.cpp.o
[ 61%] Building CXX object CMakeFiles/domoticz.dir/hardware/openzwave/control_panel/ozwcp.cpp.o
[ 61%] Building CXX object CMakeFiles/domoticz.dir/hardware/openzwave/control_panel/zwavelib.cpp.o
[ 61%] Building CXX object CMakeFiles/domoticz.dir/hardware/OTGWBase.cpp.o
[ 61%] Building CXX object CMakeFiles/domoticz.dir/hardware/OTGWSerial.cpp.o
[ 63%] Building CXX object CMakeFiles/domoticz.dir/hardware/OTGWTCP.cpp.o
[ 63%] Building CXX object CMakeFiles/domoticz.dir/hardware/PanasonicTV.cpp.o
[ 63%] Building CXX object CMakeFiles/domoticz.dir/hardware/P1MeterBase.cpp.o
[ 63%] Building CXX object CMakeFiles/domoticz.dir/hardware/P1MeterSerial.cpp.o
[ 63%] Building CXX object CMakeFiles/domoticz.dir/hardware/P1MeterTCP.cpp.o
[ 64%] Building CXX object CMakeFiles/domoticz.dir/hardware/PhilipsHue/PhilipsHue.cpp.o
[ 64%] Building CXX object CMakeFiles/domoticz.dir/hardware/PhilipsHue/PhilipsHueHelper.cpp.o
[ 64%] Building CXX object CMakeFiles/domoticz.dir/hardware/PhilipsHue/PhilipsHueSensors.cpp.o
[ 64%] Building CXX object CMakeFiles/domoticz.dir/hardware/PhilipsHue/PhilipsHueV2Sensors.cpp.o
[ 64%] Building CXX object CMakeFiles/domoticz.dir/hardware/PiFace.cpp.o
[ 66%] Building CXX object CMakeFiles/domoticz.dir/hardware/Pinger.cpp.o
[ 66%] Building CXX object CMakeFiles/domoticz.dir/hardware/PVOutput_Input.cpp.o
[ 66%] Building CXX object CMakeFiles/domoticz.dir/hardware/RAVEn.cpp.o
[ 66%] Building CXX object CMakeFiles/domoticz.dir/hardware/Rego6XXSerial.cpp.o
[ 66%] Building CXX object CMakeFiles/domoticz.dir/hardware/RelayNet.cpp.o
[ 67%] Building CXX object CMakeFiles/domoticz.dir/hardware/RFLinkBase.cpp.o
[ 67%] Building CXX object CMakeFiles/domoticz.dir/hardware/RFLinkSerial.cpp.o
[ 67%] Building CXX object CMakeFiles/domoticz.dir/hardware/RFLinkTCP.cpp.o
[ 67%] Building CXX object CMakeFiles/domoticz.dir/hardware/RFLinkMQTT.cpp.o
[ 67%] Building CXX object CMakeFiles/domoticz.dir/hardware/RFXBase.cpp.o
[ 69%] Building CXX object CMakeFiles/domoticz.dir/hardware/RFXComSerial.cpp.o
[ 69%] Building CXX object CMakeFiles/domoticz.dir/hardware/RFXComTCP.cpp.o
[ 69%] Building CXX object CMakeFiles/domoticz.dir/hardware/Rtl433.cpp.o
[ 69%] Building CXX object CMakeFiles/domoticz.dir/hardware/S0MeterBase.cpp.o
[ 69%] Building CXX object CMakeFiles/domoticz.dir/hardware/S0MeterSerial.cpp.o
[ 70%] Building CXX object CMakeFiles/domoticz.dir/hardware/S0MeterTCP.cpp.o
[ 70%] Building CXX object CMakeFiles/domoticz.dir/hardware/SatelIntegra.cpp.o
[ 70%] Building CXX object CMakeFiles/domoticz.dir/hardware/SBFSpot.cpp.o
[ 70%] Building CXX object CMakeFiles/domoticz.dir/hardware/serial/serial.cpp.o
[ 70%] Building CXX object CMakeFiles/domoticz.dir/hardware/serial/impl/unix.cpp.o
[ 72%] Building CXX object CMakeFiles/domoticz.dir/hardware/SolarEdgeAPI.cpp.o
[ 72%] Building CXX object CMakeFiles/domoticz.dir/hardware/SolarMaxTCP.cpp.o
[ 72%] Building CXX object CMakeFiles/domoticz.dir/hardware/Sterbox.cpp.o
[ 72%] Building CXX object CMakeFiles/domoticz.dir/hardware/SysfsGpio.cpp.o
[ 73%] Building CXX object CMakeFiles/domoticz.dir/hardware/Tado.cpp.o
[ 73%] Building CXX object CMakeFiles/domoticz.dir/hardware/TE923.cpp.o
[ 73%] Building CXX object CMakeFiles/domoticz.dir/hardware/TE923Tool.cpp.o
[ 73%] Building CXX object CMakeFiles/domoticz.dir/hardware/TeleinfoBase.cpp.o
[ 73%] Building CXX object CMakeFiles/domoticz.dir/hardware/TeleinfoSerial.cpp.o
[ 75%] Building CXX object CMakeFiles/domoticz.dir/hardware/TeleinfoTCP.cpp.o
[ 75%] Building CXX object CMakeFiles/domoticz.dir/hardware/Tellstick.cpp.o
[ 75%] Building CXX object CMakeFiles/domoticz.dir/hardware/TellstickFactory.cpp.o
[ 75%] Building CXX object CMakeFiles/domoticz.dir/hardware/ToonThermostat.cpp.o
[ 75%] Building CXX object CMakeFiles/domoticz.dir/hardware/TTNMQTT.cpp.o
[ 76%] Building CXX object CMakeFiles/domoticz.dir/hardware/USBtin.cpp.o
[ 76%] Building CXX object CMakeFiles/domoticz.dir/hardware/USBtin_MultiblocV8.cpp.o
[ 76%] Building CXX object CMakeFiles/domoticz.dir/hardware/VisualCrossing.cpp.o
[ 76%] Building CXX object CMakeFiles/domoticz.dir/hardware/VolcraftCO20.cpp.o
[ 76%] Building CXX object CMakeFiles/domoticz.dir/hardware/Winddelen.cpp.o
[ 77%] Building CXX object CMakeFiles/domoticz.dir/hardware/WOL.cpp.o
[ 77%] Building CXX object CMakeFiles/domoticz.dir/hardware/Wunderground.cpp.o
[ 77%] Building CXX object CMakeFiles/domoticz.dir/hardware/XiaomiGateway.cpp.o
[ 77%] Building CXX object CMakeFiles/domoticz.dir/hardware/XiaomiDeviceSupport.cpp.o
[ 77%] Building CXX object CMakeFiles/domoticz.dir/hardware/Yeelight.cpp.o
[ 79%] Building CXX object CMakeFiles/domoticz.dir/hardware/YouLess.cpp.o
[ 79%] Building CXX object CMakeFiles/domoticz.dir/hardware/ZiBlueBase.cpp.o
[ 79%] Building CXX object CMakeFiles/domoticz.dir/hardware/ZiBlueSerial.cpp.o
[ 79%] Building CXX object CMakeFiles/domoticz.dir/hardware/ZiBlueTCP.cpp.o
[ 79%] Building CXX object CMakeFiles/domoticz.dir/hardware/ZWaveBase.cpp.o
[ 80%] Building CXX object CMakeFiles/domoticz.dir/hardware/plugins/DelayedLink.cpp.o
[ 80%] Building CXX object CMakeFiles/domoticz.dir/hardware/plugins/Plugins.cpp.o
[ 80%] Building CXX object CMakeFiles/domoticz.dir/hardware/plugins/PluginManager.cpp.o
[ 80%] Building CXX object CMakeFiles/domoticz.dir/hardware/plugins/PluginProtocols.cpp.o
[ 80%] Building CXX object CMakeFiles/domoticz.dir/hardware/plugins/PluginTransports.cpp.o
[ 82%] Building CXX object CMakeFiles/domoticz.dir/hardware/plugins/PythonObjects.cpp.o
[ 82%] Building CXX object CMakeFiles/domoticz.dir/hardware/plugins/PythonObjectEx.cpp.o
[ 82%] Building CXX object CMakeFiles/domoticz.dir/notifications/NotificationBase.cpp.o
[ 82%] Building CXX object CMakeFiles/domoticz.dir/notifications/NotificationBrowser.cpp.o
[ 82%] Building CXX object CMakeFiles/domoticz.dir/notifications/NotificationEmail.cpp.o
[ 83%] Building CXX object CMakeFiles/domoticz.dir/notifications/NotificationFCM.cpp.o
[ 83%] Building CXX object CMakeFiles/domoticz.dir/notifications/NotificationHelper.cpp.o
[ 83%] Building CXX object CMakeFiles/domoticz.dir/notifications/NotificationHTTP.cpp.o
[ 83%] Building CXX object CMakeFiles/domoticz.dir/notifications/NotificationKodi.cpp.o
[ 85%] Building CXX object CMakeFiles/domoticz.dir/notifications/NotificationLogitechMediaServer.cpp.o
[ 85%] Building CXX object CMakeFiles/domoticz.dir/notifications/NotificationPushbullet.cpp.o
[ 85%] Building CXX object CMakeFiles/domoticz.dir/notifications/NotificationProwl.cpp.o
[ 85%] Building CXX object CMakeFiles/domoticz.dir/notifications/NotificationPushover.cpp.o
[ 85%] Building CXX object CMakeFiles/domoticz.dir/notifications/NotificationPushsafer.cpp.o
[ 86%] Building CXX object CMakeFiles/domoticz.dir/notifications/NotificationPushalot.cpp.o
[ 86%] Building CXX object CMakeFiles/domoticz.dir/notifications/NotificationSMS.cpp.o
[ 86%] Building CXX object CMakeFiles/domoticz.dir/notifications/NotificationTelegram.cpp.o
[ 86%] Building CXX object CMakeFiles/domoticz.dir/smtpclient/SMTPClient.cpp.o
[ 86%] Building CXX object CMakeFiles/domoticz.dir/tcpserver/TCPClient.cpp.o
[ 88%] Building CXX object CMakeFiles/domoticz.dir/tcpserver/TCPServer.cpp.o
[ 88%] Building CXX object CMakeFiles/domoticz.dir/webserver/Base64.cpp.o
[ 88%] Building CXX object CMakeFiles/domoticz.dir/webserver/connection.cpp.o
[ 88%] Building CXX object CMakeFiles/domoticz.dir/webserver/connection_manager.cpp.o
[ 88%] Building CXX object CMakeFiles/domoticz.dir/webserver/cWebem.cpp.o
[ 89%] Building CXX object CMakeFiles/domoticz.dir/webserver/fastcgi.cpp.o
[ 89%] Building CXX object CMakeFiles/domoticz.dir/webserver/mime_types.cpp.o
[ 89%] Building CXX object CMakeFiles/domoticz.dir/webserver/reply.cpp.o
[ 89%] Building CXX object CMakeFiles/domoticz.dir/webserver/request_handler.cpp.o
[ 89%] Building CXX object CMakeFiles/domoticz.dir/webserver/request_parser.cpp.o
[ 91%] Building CXX object CMakeFiles/domoticz.dir/webserver/server.cpp.o
[ 91%] Building CXX object CMakeFiles/domoticz.dir/webserver/Websockets.cpp.o
[ 91%] Building CXX object CMakeFiles/domoticz.dir/webserver/WebsocketHandler.cpp.o
[ 91%] Building CXX object CMakeFiles/domoticz.dir/tinyxpath/action_store.cpp.o
[ 91%] Building CXX object CMakeFiles/domoticz.dir/tinyxpath/htmlutil.cpp.o
[ 92%] Building CXX object CMakeFiles/domoticz.dir/tinyxpath/lex_util.cpp.o
[ 92%] Building CXX object CMakeFiles/domoticz.dir/tinyxpath/node_set.cpp.o
[ 92%] Building CXX object CMakeFiles/domoticz.dir/tinyxpath/tinystr.cpp.o
[ 92%] Building CXX object CMakeFiles/domoticz.dir/tinyxpath/tinyxml.cpp.o
[ 92%] Building CXX object CMakeFiles/domoticz.dir/tinyxpath/tinyxmlerror.cpp.o
[ 94%] Building CXX object CMakeFiles/domoticz.dir/tinyxpath/tinyxmlparser.cpp.o
[ 94%] Building CXX object CMakeFiles/domoticz.dir/tinyxpath/tokenlist.cpp.o
[ 94%] Building CXX object CMakeFiles/domoticz.dir/tinyxpath/xml_util.cpp.o
[ 94%] Building CXX object CMakeFiles/domoticz.dir/tinyxpath/xpath_expression.cpp.o
[ 95%] Building CXX object CMakeFiles/domoticz.dir/tinyxpath/xpath_processor.cpp.o
[ 95%] Building CXX object CMakeFiles/domoticz.dir/tinyxpath/xpath_stream.cpp.o
[ 95%] Building CXX object CMakeFiles/domoticz.dir/tinyxpath/xpath_stack.cpp.o
[ 95%] Building CXX object CMakeFiles/domoticz.dir/tinyxpath/xpath_static.cpp.o
[ 95%] Building CXX object CMakeFiles/domoticz.dir/tinyxpath/xpath_syntax.cpp.o
[ 97%] Linking CXX executable domoticz
[ 97%] Built target domoticz
[ 97%] Building CXX object CMakeFiles/domoticztester.dir/cmake_pch.hxx.gch
[ 97%] Building CXX object CMakeFiles/domoticztester.dir/main/stdafx.cpp.o
[ 97%] Building CXX object CMakeFiles/domoticztester.dir/main/Helper.cpp.o
[ 97%] Building CXX object CMakeFiles/domoticztester.dir/main/CmdLine.cpp.o
[ 98%] Building CXX object CMakeFiles/domoticztester.dir/main/domoticz_tester.cpp.o
[ 98%] Building CXX object CMakeFiles/domoticztester.dir/main/BaroForecastCalculator.cpp.o
[ 98%] Building CXX object CMakeFiles/domoticztester.dir/main/HTMLSanitizer.cpp.o
[ 98%] Building CXX object CMakeFiles/domoticztester.dir/main/localtime_r.cpp.o
[ 98%] Building CXX object CMakeFiles/domoticztester.dir/main/SunRiseSet.cpp.o
[100%] Building CXX object CMakeFiles/domoticztester.dir/main/TrendCalculator.cpp.o
[100%] Building CXX object CMakeFiles/domoticztester.dir/main/WindCalculation.cpp.o
[100%] Building CXX object CMakeFiles/domoticztester.dir/main/json_helper.cpp.o
[100%] Building CXX object CMakeFiles/domoticztester.dir/hardware/ColorSwitch.cpp.o
[100%] Linking CXX executable domoticztester
[100%] Built target domoticztester
```
